tensorflow/core/kernels/data/map_dataset_op_test.cc

```diff
@@ -53,526 +54,682 @@ class MapDatasetOpTest : public DatasetOpsTestBase {
   }
 };
 
-struct TestCase {
-  int64 start;
-  int64 end;
-  int64 step;
+class MapDatasetParams : public DatasetParams {
+ public:
+  MapDatasetParams(int64 start, int64 stop, int64 step,
+                   std::vector<Tensor> other_arguments,
+                   FunctionDefHelper::AttrValueWrapper func,
+                   std::vector<FunctionDef> func_lib,
+                   DataTypeVector type_arguments, DataTypeVector output_dtypes,
+                   std::vector<PartialTensorShape> output_shapes,
+                   bool use_inter_op_parallelism, bool preserve_cardinality,
+                   string node_name)
+      : DatasetParams(std::move(output_dtypes), std::move(output_shapes),
+                      std::move(node_name)),
+        range_dataset_params(start, stop, step, {DT_INT64},
+                             {PartialTensorShape({})}, ""),
+        other_arguments(std::move(other_arguments)),
+        func(std::move(func)),
+        func_lib(std::move(func_lib)),
+        type_arguments(std::move(type_arguments)),
+        use_inter_op_parallelism(use_inter_op_parallelism),
+        preserve_cardinality(preserve_cardinality) {}
+
+  Status MakeInputs(gtl::InlinedVector<TensorValue, 4>* inputs) override {
+    if (input_dataset.NumElements() == 0 ||
+        input_dataset.dtype() != DT_VARIANT) {
+      return tensorflow::errors::Internal(
+          "The input dataset is not populated as the dataset tensor yet.");
+    }
+    *inputs = {TensorValue(&input_dataset)};
+    for (auto& argument : other_arguments) {
+      inputs->emplace_back(TensorValue(&argument));
+    }
+    return Status::OK();
+  }
+
+  RangeDatasetParams range_dataset_params;
+  Tensor input_dataset;
+  std::vector<Tensor> other_arguments;
   FunctionDefHelper::AttrValueWrapper func;
   std::vector<FunctionDef> func_lib;
-  std::vector<Tensor> expected_outputs;
-  DataTypeVector expected_output_dtypes;
-  std::vector<PartialTensorShape> expected_output_shapes;
-  int64 expected_cardinality;
-  std::vector<int> breakpoints;
+  DataTypeVector type_arguments;
+  bool use_inter_op_parallelism;
+  bool preserve_cardinality;
 };
 
-TestCase TestCase1() {
-  return {/*start*/ 0,
-          /*end*/ 10,
-          /*step*/ 3,
-          /*func*/
+MapDatasetParams MapDataset1() {
+  return {/*start=*/0,
```

Comment:
Why are the map dataset tests using different `start` and `step`? I fee like the test should (similar to `batch` test) simply use `num_input_elements` and leave the testing of different start / stop / step behaviors to the `range` test.

Reply:
It will help understand the input value for the map function by using start/stop/step. Will `num_input_elements` make the code not easy to read?